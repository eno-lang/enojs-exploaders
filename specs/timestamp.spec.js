const eno = require('enojs');
const loaders = require('../loaders.js');

describe('datetime', () => {

  let document;

  describe('1997', () => {
    it('returns a correctly set Date object', () => {
      document = eno.parse('datetime: 1997');
      expect(document.field('datetime', loaders.datetime)).toMatchSnapshot();
    });
  });

  describe('1997-07', () => {
    it('returns a correctly set Date object', () => {
      document = eno.parse('datetime: 1997-07');
      expect(document.field('datetime', loaders.datetime)).toMatchSnapshot();
    });
  });

  describe('1997-07-16', () => {
    it('returns a correctly set Date object', () => {
      document = eno.parse('datetime: 1997-07-16');
      expect(document.field('datetime', loaders.datetime)).toMatchSnapshot();
    });
  });

  describe('1997-07-16T19:20+01:00', () => {
    it('returns a correctly set Date object', () => {
      document = eno.parse('datetime: 1997-07-16T19:20+01:00');
      expect(document.field('datetime', loaders.datetime)).toMatchSnapshot();
    });
  });

  describe('1997-07-16T19:20:30+01:00', () => {
    it('returns a correctly set Date object', () => {
      document = eno.parse('datetime: 1997-07-16T19:20:30+01:00');
      expect(document.field('datetime', loaders.datetime)).toMatchSnapshot();
    });
  });

  describe('1997-07-16T19:20:30.45+01:00', () => {
    it('returns a correctly set Date object', () => {
      document = eno.parse('datetime: 1997-07-16T19:20:30.45+01:00');
      expect(document.field('datetime', loaders.datetime)).toMatchSnapshot();
    });
  });

  describe('1994-11-05T08:15:30-05:00', () => {
    it('returns a correctly set Date object', () => {
      document = eno.parse('datetime: 1994-11-05T08:15:30-05:00');
      expect(document.field('datetime', loaders.datetime)).toMatchSnapshot();
    });
  });

  describe('1994-11-05T13:15:30Z', () => {
    it('returns a correctly set Date object', () => {
      document = eno.parse('datetime: 1994-11-05T13:15:30Z');
      expect(document.field('datetime', loaders.datetime)).toMatchSnapshot();
    });
  });

  describe('2002 12 14', () => {
    it('throws an error', () => {
      document = eno.parse('datetime: 2002 12 14');
      expect(() => document.field('datetime', loaders.datetime)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('January', () => {
    it('throws an error', () => {
      document = eno.parse('datetime: January');
      expect(() => document.field('datetime', loaders.datetime)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('13:00', () => {
    it('throws an error', () => {
      document = eno.parse('datetime: 13:00');
      expect(() => document.field('datetime', loaders.datetime)).toThrowErrorMatchingSnapshot();
    });
  });
});
