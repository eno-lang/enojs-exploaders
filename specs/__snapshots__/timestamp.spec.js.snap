// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datetime 13:00 throws an error 1`] = `
"'datetime' must be a datetime (e.g. 1997-07-16, 1994-11-05T13:15:30Z, see https://www.w3.org/TR/NOTE-datetime)

   Line | Content
 >    1 | datetime: 13:00
"
`;

exports[`datetime 1994-11-05T08:15:30-05:00 returns a correctly set Date object 1`] = `1994-11-05T13:15:30.000Z`;

exports[`datetime 1994-11-05T13:15:30Z returns a correctly set Date object 1`] = `1994-11-05T13:15:30.000Z`;

exports[`datetime 1997 returns a correctly set Date object 1`] = `1997-01-01T00:00:00.000Z`;

exports[`datetime 1997-07 returns a correctly set Date object 1`] = `1997-07-01T00:00:00.000Z`;

exports[`datetime 1997-07-16 returns a correctly set Date object 1`] = `1997-07-16T00:00:00.000Z`;

exports[`datetime 1997-07-16T19:20+01:00 returns a correctly set Date object 1`] = `1997-07-16T18:20:00.000Z`;

exports[`datetime 1997-07-16T19:20:30+01:00 returns a correctly set Date object 1`] = `1997-07-16T18:20:30.000Z`;

exports[`datetime 1997-07-16T19:20:30.45+01:00 returns a correctly set Date object 1`] = `1997-07-16T18:20:30.450Z`;

exports[`datetime 2002 12 14 throws an error 1`] = `
"'datetime' must be a datetime (e.g. 1997-07-16, 1994-11-05T13:15:30Z, see https://www.w3.org/TR/NOTE-datetime)

   Line | Content
 >    1 | datetime: 2002 12 14
"
`;

exports[`datetime January throws an error 1`] = `
"'datetime' must be a datetime (e.g. 1997-07-16, 1994-11-05T13:15:30Z, see https://www.w3.org/TR/NOTE-datetime)

   Line | Content
 >    1 | datetime: January
"
`;
